xarray/core/common.py

```diff
@@ -748,6 +748,25 @@ def groupby_bins(
             },
         )
 
+    def weighted(self, weights):
+        """
+        Weighted operations.
+
+        Parameters
+        ----------
+        weights : DataArray
+            An array of weights associated with the values in this Dataset.
+            Each value in the data contributes to the reduction operation
+            according to its associated weight.
+
+        Notes
+        -----
+        ``weights`` must be a :py:class:`DataArray` and cannot contain missing values.
```

Comment:
`py:class` also in source files? If so there would be more instances to update.

Reply:
we are very inconsistent about this AFAIR, so it should be fine.

Reply:
Not a super strong view, but I would vote for not having these in docstrings: I'd guess they're looked at just as much in raw form as in the online documentation. And in this case, I'm not sure how helpful having something as fundamental as `DataArray` linked is.

Reply:
fair point